modules/task.js

```diff
@@ -0,0 +1,96 @@
+export default class Task {
+  constructor() {
+    this.lists = JSON.parse(localStorage.getItem('list')) || [];
+    this.editId = null;
+  }
+
+  addList() {
+    const taskInput = document.getElementById('task-input');
+    const list = { description: '', completed: false, index: 0 };
+    list.description = taskInput.value;
+    if (!list.description) {
+      return null;
+    } if (this.editId) {
+      const index = this.editId[this.editId.length - 1];
+      this.lists.splice(index, 1, list);
+      localStorage.setItem('list', JSON.stringify(this.lists));
+    } else {
+      list.index = this.lists.length + 1;
+      this.lists.push(list);
+      localStorage.setItem('list', JSON.stringify(this.lists));
```

Comment:
localStorage.setItem('list', JSON.stringify(this.lists)); statement is repreatedly used. and violates DRY best practices. Kindly restructure your code.